client/src/app/components/page/task-detail/task-detail.component.ts

```diff
@@ -68,12 +74,14 @@ export class TaskDetailComponent implements OnInit, OnDestroy {
                 description: this.taskForm.value.description,
                 createdBy: this.as.currentUserValue.userName,
                 parentId: null,
-                tags: this.taskForm.value.tags
+                tags: this.taskForm.value.tags,
+                deadline: this.taskForm.value.deadline
             }
             this.loaderService.setLoading(true);
             this.ts.addTask(newTask).pipe(
                 finalize(() => {
                     this.loaderService.setLoading(false);
+                    this.fetchedData = true;
```

Comment:
? what this is for

Reply:
I use it here
```
<app-date-picker *ngIf="fetchedData || !updateMode" [pickedDate]="taskForm.value.deadline"
                (onDatePicked)="onDateChange($event)"></app-date-picker>
```
to pass value after fetching from the server

Reply:
please left TODO: refactor this

Reply:
Done